### PR TITLE
Add Easy Coding Standard + apply PSR-2 + few common sets

### PR DESCRIPTION
**Pull request by TomasVotruba**
Follow up to coding standard PRs #14741, #14738 

## This PR

- adds [Easy Coding Standard](https://github.com/symplify/easyCodingStandard/) - a tool that uses **PHP_CodeSniffer + PHP CS Fixer**; **both have many rules that the other misses**, this way you can use any sniff or fixer from each family
- adds http://github.com/phpmyadmin/coding-standard/ right to `ecs.yml` config
- applies PSR-2 set to the code
- adds Travis run
- add composer script

Current PHP_CodeSniffer didn't even load the phpmyadmin/coding-standard:

```
vendor/bin/phpcs --exclude=PSR1.Files.SideEffects,Generic.Metrics.NestingLevel -i
// The installed coding standards are PSR12, PSR2, Zend, PSR1, MySource, PEAR and Squiz
```

- this is now fixed, since PR moves the `phpmyadmin/coding-standard` to `ecs.yml`
- local `phpcs.xml` moved to `ecs.yml`
- removes old config + PHP_CodeSniffer setup + spamming `@codingStandardIgnore` annotatoin - it's integrated in ECS + `ecs.yml` now

You can read more about [PHP_CodeSniffer to ECS migration](https://www.tomasvotruba.cz/blog/2018/06/04/how-to-migrate-from-php-code-sniffer-to-easy-coding-standard/)

## How to use

Dry run:

```
composer cs
```

Fix run:

```
composer fix-cs
```

Show what loaded fixers/snfifers:

```
vendor/bin/ecs show
```

### Other Unrelated Changes

- speedup Travis process by disabling xdebug, when not needed

### CODE REVIEW

**Comment by TomasVotruba** [comment]
## WIP - I'm waiting for the Travis to finish to fix found bugs, no need to review now

**Comment by mauriciofauth** [comment]
@TomasVotruba PHP_CodeSniffer automatically loads the standard from the configuration file.
See: https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#using-a-default-configuration-file

You can check it by running: `vendor/bin/phpcs -v`

**Comment by TomasVotruba** [comment]
I understand that, but why it's not displayed on using `-i`?

```
// The installed coding standards are PSR12, PSR2, Zend, PSR1, MySource, PEAR and Squiz
```

There should be "phpMyAdmin Standard" - https://github.com/phpmyadmin/coding-standard/blob/8cf1c841460216f5ccf186c5f42bf187ad125b5b/PMAStandard/ruleset.xml#L2

**Comment by TomasVotruba** [comment]
Tests are passing now (apart that PHP 7.2 downloading mariadb fail :smile: ).
I've cleaned up some configuraiton code - take that as a suggestion.

It's ready to review and feedback.

File: README.rst [hunk 1]
```diff
@@ -19,9 +19,6 @@ Code status
 .. image:: https://codecov.io/gh/phpmyadmin/phpmyadmin/branch/master/graph/badge.svg
     :target: https://codecov.io/gh/phpmyadmin/phpmyadmin
 
-.. image:: https://scrutinizer-ci.com/g/phpmyadmin/phpmyadmin/badges/quality-score.png?s=93dfde29ffa5771d9c254b7ffb11c4e673315035
-    :target: https://scrutinizer-ci.com/g/phpmyadmin/phpmyadmin/
```
Comment (mauriciofauth) [inline README.rst]: Why did you remove Scrutinizer?

File: build.xml [hunk 2]
```diff
@@ -157,5 +77,5 @@
   </apply>
  </target>
 
- <target name="build" depends="clean,phpunit,pdepend,phpmd,phpcpd,phpcs,phpdoc,phploc,phpcb,lint,jshint,locales"/>
```
Comment (mauriciofauth) [inline build.xml]: Why did you remove these tools?

**Comment by mauriciofauth** [comment]
> I understand that, but why it's not displayed on using -i?

Because it's not installed. You can install it if you want, but you don't have to.

**Comment by williamdes** [comment]
I think some cleanup should be moved to another PR and the config in this one.
@mauriciofauth Agree ?

**Comment by mauriciofauth** [comment]
I agree with @williamdes.

File: README.rst (hunk 1 above)
Comment (mauriciofauth) [reply README.rst]: > remove .scrutinizer.yml, since it's better covered by newly added PHPStan - also report is more clear (on this note: https://www.tomasvotruba.cz/blog/2017/01/28/why-I-switched-scrutinizer-for-phpstan-and-you-should-too/)

I just saw it.

**Comment by mauriciofauth** [comment]
@TomasVotruba Most coding style fixes are welcome and easy to review and merge, but other changes need more review and discussion before they are applied. So it would be better to separate the changes into more specific PRs.

File: build.xml (hunk 2 above)
Comment (TomasVotruba) [reply build.xml]: `pdepend,phpmd,phpcpd,phpcs,phpdoc,phploc,phpcb` are not installed in this repository.

I assume they're not used. If you use them, they should be part of `composer.json` `"require-dev" section, since they don't work at the moment.

I checked `.travis.yml` as well, but I didn't find them. But there is lot of new code for me, so I can be wrong.

**Comment by TomasVotruba** [comment]
Roger that! I'll seperate it :+1:

File: build.xml (hunk 2 above)
Comment (mauriciofauth) [reply build.xml]: They are used in phpMyAdmin Jenkins server.

**Comment by TomasVotruba** [comment]
I don't undrestand the error in "Build docs":
https://travis-ci.org/phpmyadmin/phpmyadmin/jobs/460856814#L547

Could you help me?

**Comment by mauriciofauth** [comment]
@TomasVotruba It was fixed by be7464b60e2e785f4c4885239f600b9dce3954c0.

**Comment by TomasVotruba** [comment]
@mauriciofauth Great! That was my tip, but I wasn't sure :)
I'll rebase in a sec

**Comment by williamdes** [comment]
@TomasVotruba Is this PR only for CS ?
If it is can you revert the config modifications of https://github.com/phpmyadmin/phpmyadmin/pull/14757/commits/2e444648efbfbea3abfed4fa96355a9e665cd780 ?

**Comment by williamdes** [comment]
@TomasVotruba Good catch for the Xdebug config !

**Comment by TomasVotruba** [comment]
> @TomasVotruba Good catch for the Xdebug config !

Thanks

> @TomasVotruba Is this PR only for CS ?
> If it is can you revert the config modifications of 2e44464 ?

Yes. What file config do you mean exactly? The link goes to whole commit.

If you mean this: 

```diff
- composer phpcs
+ composer cs
```

It's on purpose, because there are about 10 places where PHPStan, PSR-2 including Fixer rules and old phpcs are in conflict. Mainly in `@param Type[] ...$variadic`. 

This PR adds new style that PSR-2 and PHPStan respects.

**Comment by TomasVotruba** [comment]
I just rebased on master + resolved the conflict

File: composer.json [hunk 3]
```diff
@@ -86,7 +86,8 @@
         "samyoul/u2f-php-server": "^1.1",
         "phpmyadmin/coding-standard": "^0.3",
         "phpstan/phpstan": "^0.10.3",
-        "facebook/webdriver": "^1.6"
+        "facebook/webdriver": "^1.6",
+        "symplify/easy-coding-standard": "^5.2"
```
Comment (mauriciofauth) [inline composer.json]: I'm not sure about replacing PHP_CodeSniffer with Easy Coding Standard.
phpMyAdmin doesn't have a strict coding standard yet and so phpcs is serving our current needs very well.
I'm more inclined to improve our [phpcs coding standard](https://github.com/phpmyadmin/coding-standard) than to change tool.

Comment (TomasVotruba) [reply composer.json]: It **uses** PHP_CodeSnfifer, so it's not about replacing, but rather opening to new rules.

Having coding standard in standalone repository makes only sense in case of you have some custom sniffs/fixers.

<br>

Here are few big PHP projects that use it, if you want to look:

- https://github.com/SyliusLabs/CodingStandard
- https://github.com/nette/coding-standard
- https://github.com/php-ai/php-ml/blob/master/ecs.yml

Also, old implementation missed many PSR-2 violations, see this commit: https://github.com/phpmyadmin/phpmyadmin/pull/14757/commits/d8337b85db108aa7752d87449e723c51627ee2ae

Comment (mauriciofauth) [reply composer.json]: Current phpMyAdmin coding standard does have custom sniffs and it does not require PSR-2. See: https://github.com/phpmyadmin/coding-standard/tree/0.3
What we have on _master_ is not finished yet.

Comment (TomasVotruba) [reply composer.json]: It's possible they're not needed. Concat sniff is already covered by existing binary fixer, for file level sniff there is header fixer. The other I don't quite get. 

The PSR2 was run by default here. That's why I had to disable it in Travis.

Anyway, ECS can be used by with custom standards as well: 
https://github.com/nette/coding-standard

Comment (TomasVotruba) [reply composer.json]: > What we have on master is not finished yet.

Having coding standard in main project prevents such dissonance.

Comment (mauriciofauth) [reply composer.json]: > Having coding standard in standalone repository makes only sense in case of you have some custom sniffs/fixers.

> Having coding standard in main project prevents such dissonance.

We also plan to use the coding standard in other phpMyAdmin projects. So a standalone repository makes sense for that.

> It's possible they're not needed

That's why they were recently replaced in the master.

**Comment by TomasVotruba** [comment]
Any feedback on this?
 
Are there any drawbacks that I could explain better/resolve so it would be accepted?

File: composer.json (hunk 3 above)
Comment (mauriciofauth) [reply composer.json]: Sorry for the delay in answering. In the future we will improve our [coding standard](https://github.com/phpmyadmin/coding-standard) and make it more strict and then we will keep an eye on Easy Coding Standard as it seems to be a great tool, but for now we prefer to keep only PHP_CodeSniffer.

**Comment by mauriciofauth** [comment]
@TomasVotruba Could you please only keep the coding style fixes in this pull request?

File: composer.json (hunk 3 above)
Comment (TomasVotruba) [reply composer.json]: That was clear to me from previous comments. I'm curious why.

**Comment by TomasVotruba** [comment]
@mauriciofauth That's not possible, since many fixed are missing in your current standard definition.

File: composer.json (hunk 3 above)
Comment (TomasVotruba) [reply composer.json]: Don't take me wrong please. I've just invested many hours and fixing various conflicts, rebasing again on master and resolve issues on Travis, that were unrelated.

I expect such message in the start of such PR - other repositories do that -, before I invest such big amount of work and free time. 

I can't find any reason, apart...

> phpcs is serving our current needs very well.

...which is really unclear to me. Why coding standard can't be improved?

Comment (mauriciofauth) [reply composer.json]: @TomasVotruba I'm sorry, I may not be expressing myself right, English is not my native language. We greatly appreciate the work of our contributors.

We want to improve our [coding standard](https://github.com/phpmyadmin/coding-standard) first, and then apply it in this repository, not the opposite. What I wanted to say was we want to keep phpcs in this repository as long as our standard is not improved.

> phpcs is serving our current needs very well.

What I meant by that is that I see no benefit in switching tools if we stick to the same rules. That's why I said that instead I prefer to improve our [coding standard](https://github.com/phpmyadmin/coding-standard) and we would be grateful if you could help us improve it.

I hope I expressed myself better this time. :smile:

Comment (TomasVotruba) [reply composer.json]: Hi, I understand that before.
That why I replied: https://github.com/phpmyadmin/phpmyadmin/pull/14757#discussion_r238825831

Comment (TomasVotruba) [reply composer.json]: I posted that after no-clear reply on my comments :D 

So will you be open to try ECS there if that would lead to CS improvement?

Comment (mauriciofauth) [reply composer.json]: @TomasVotruba Yes.

**Comment by TomasVotruba** [comment]
I do not have time and motivation to work on this at the moment. So I'm closing it. 

Anyone feel free to re-use my work and continue.
